### ISSUE TITLE
Slow page loads in JIRA when CODE block contains "(?, ?)"

### ISSUE THREAD

**Issue by ganela** [catawiki/Prism#5]
We found that if a JIRA ticket has a {code} block that contains a large amount of question marks in it, it takes a while for the page to render. For example, we had a user that added some SQL Log data to a {code} block that contained over 9700 instances of "(?, ?)" (See snippet below) and the page took ~7 minutes to load. We disabled the PRISM Add-on, and the page loaded instantly. 

We're on...
JIRA Version: Server 7.6.7
PRISM Version: 1.2.3

**Here's a snippet of the offending code:**

NOTE: This is just a small snippet of the code that was causing the page to load slowly. As I previously mentioned, there are actually ~9700 instances of "(?, ?)" in the full version of the code.

> 2018-09-04 17:05:18] report.CRITICAL: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '4613' for key 'PRIMARY', query was: INSERT  INTO `search_tmp_5b8ebb4e994da5_88027289` (`entity_id`,`score`) VALUES (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?), (?, ?) [] []

**Workaround**: To remedy the problem we converted the {code} block to a {panel} block.

**Comment by idsvandermolen** [comment]
Did you use a regular `{code}` block or did you specify a language (like `{code:sql}`)? It is quite possible that Prism would struggle rendering the SQL syntax using the default language (`ruby`).

**Comment by siavashs** [comment]
I could not reproduce this issue with the mentioned snippet and either `java`, `ruby` or `sql` languages set.

**Comment by ganela** [comment]
> Did you use a regular `{code}` block or did you specify a language (like `{code:sql}`)? It is quite possible that Prism would struggle rendering the SQL syntax using the default language (`ruby`).

Hi @idsvandermolen - The code tag did not have a language specified.

**Comment by ganela** [comment]
> I could not reproduce this issue with the mentioned snippet and either `java`, `ruby` or `sql` languages set.
> The test was done on _7.12.x_, will test with _7.6.x_ but I expect the same results since Prism is just a client JS library.

Hi @siavashs  - As I previously mentioned, the sample above is just a snippet of the code block. The actual code block contains ~9700 instances of "(?, ?),". Please try duplicating it with [this sample](https://github.com/catawiki/Prism/files/2425473/PRISM-CODE.BLOCK.ISSUE-SAMPLE.txt) that contains the actual code block causing the issue.

**Comment by siavashs** [comment]
I checked your sample code (Even gedit struggles to render this text on my laptop).
JIRA doesn't accept such a huge text by default:
>The entered text is too long. It exceeds the allowed limit of 32767 characters.

I can reproduce the issue by setting unsafe limits. It takes `55.79s` for Firefox to render this code block (JIRA is running on the same machine which is a laptop), Firefox notified me multiple times during the page loading if I want to stop it.

I captured the performance, it's mostly **Recalculate Style** and **GC**.

I'm not going to invest more time in this since:
- you are using unsafe limits
- this is not a normal use case
- it's a performance issue on PrismJS library that we use
